x/application/keeper/msg_server_undelegate_from_gateway.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+
+	sdkerrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/pokt-network/poktroll/x/application/types"
+)
+
+func (k msgServer) UndelegateFromGateway(goCtx context.Context, msg *types.MsgUndelegateFromGateway) (*types.MsgUndelegateFromGatewayResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	logger := k.Logger().With("method", "UndelegateFromGateway")
+	logger.Info(fmt.Sprintf("About to undelegate application from gateway with msg: %v", msg))
+
+	if err := msg.ValidateBasic(); err != nil {
+		logger.Error(fmt.Sprintf("Undelegation Message failed basic validation: %v", err))
```

Comment:
`// TODO_TECHDEBT(#348): ensure consistent error convention is used.`

Reply:
See [discord#code-reviews](https://discord.com/channels/824324475256438814/1138521103338328258/1209125996918480906).